app/controllers/CyaSupplementaryUnitController.scala

```diff
@@ -93,7 +93,10 @@ class CyaSupplementaryUnitController @Inject() (
             model.supplementaryUnit.getOrElse("")
```

Comment:
(slightly above) why are these things being defined together rather than separately. 
val initialHasSuppUnitOpt = request.session.get(initialValueOfHasSuppUnit).map(_.toBoolean)
val initialSuppUnitOpt = request.session.get(initialValueOfSuppUnit)

I guess it works but it makes the code more complicated?

Reply:
I initially defined them separately, but it looked there were more repeated lines and val declarations just for getting the values, so simplified it :). If that makes it look complicated, then I will declare them separately.

Reply:
Ok, that's fair, although its more lines of code to do it as tuples than just variables 🙂

Reply:
Done it to be more readable.